modules/trusted-firmware-m/Kconfig

```diff
@@ -64,12 +64,41 @@ config TFM_KEY_FILE_NS
 	  firmware images.
 
 config TFM_PROFILE
-	string "The build profile used for TFM Secure image."
+	string
 	depends on BUILD_WITH_TFM
+	default "profile_small" if TFM_PROFILE_TYPE_SMALL
+	default "profile_medium" if TFM_PROFILE_TYPE_MEDIUM
+	default "profile_large" if TFM_PROFILE_TYPE_LARGE
 	help
 	  Build profile used to build tfm_s image. The available values are
-	  profile_medium and profile_small. The default profile does not need
-	  to have this configuration set.
+	  profile_large, profile_medium and profile_small. The default profile
+	  does not need to have this configuration set.
+
+choice TFM_PROFILE_TYPE
+	prompt "TF-M build profile"
+	depends on BUILD_WITH_TFM
+	default TFM_PROFILE_TYPE_NOT_SET
+	help
+	  The TF-M build profile selection. Can be empty (not set),
+	  small, medium or large. Certain profile types enable other
+	  TF-M configuration options, namely, the IPC model and the
+	  isolation level.
+
+config TFM_PROFILE_TYPE_NOT_SET
+	bool "TF-M build profile is not set"
+
+config TFM_PROFILE_TYPE_SMALL
+	bool "TF-M build profile: small"
+
+config TFM_PROFILE_TYPE_MEDIUM
+	bool "TF-M build profile: medium"
+	select TFM_IPC
```

Comment:
I think I'd prefer if this was handled in the TFM_IPC config like for the ISOLATION_LEVEL. I think it's clearer that way, and if we later make IPC the default, profile_small will still have the correct TFM_IPC setting.

Reply:
done.